src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -2562,6 +2575,52 @@ theorem udiv_self {x : BitVec w} :
       ↓reduceIte, toNat_udiv]
     rw [Nat.div_self (by omega), Nat.mod_eq_of_lt (by omega)]
 
+private theorem Nat.div_le_div_left (hcb : c ≤ b) (hc : 0 < c) : a / b ≤ a / c :=
+  (Nat.le_div_iff_mul_le hc).2 <|
+    Nat.le_trans (Nat.mul_le_mul_left _ hcb) (Nat.div_mul_le_self a b)
+
+private theorem Nat.div_add_le_right {z : Nat} (h : 0 < z) (x y : Nat) :
+    x / (y + z) ≤ x / z :=
+  Nat.div_le_div_left (by omega) h
```

Comment:
Should these become normal `Nat` theorems? Also, why are `z` and `x v` not declared in the same list of arguments.

Reply:
Since `z` can be inferred from the type of `h`, while `x` and `y` cannot. I also consciously put `h` first, since that argument will have to be provided when using this as a rewrite, while `x` and `y` may sometimes be inferred through unification

Reply:
Also, yes, these make sense as normal theorems; I moved them to `Nat.Div.Lemmas`